gemini-2/live_api_starter.py

```diff
@@ -113,6 +122,33 @@ async def get_frames(self):
         # Release the VideoCapture object
         cap.release()
 
+    def _get_screen(self, sct, monitor):
+        i = sct.grab(monitor)
+
+        mime_type = "image/jpeg"
+        image_bytes = mss.tools.to_png(i.rgb, i.size)
+        img = PIL.Image.open(io.BytesIO(image_bytes))
+
+        image_io = io.BytesIO()
+        img.save(image_io, format="jpeg")
+        image_io.seek(0)
+
+        image_bytes = image_io.read()
+        return {"mime_type": mime_type, "data": base64.b64encode(image_bytes).decode()}
+
+    async def get_screen(self):
+        sct = mss.mss()
+        monitor = sct.monitors[0]
+
+        while True:
+            frame = await asyncio.to_thread(self._get_screen, sct, monitor)
```

Comment:
@clementfarabet I'm getting that error when running on my windows PC:
| Traceback (most recent call last):
| File "C:\Users\giom\myenv_test\live_api_starter.py", line 146, in get_screen
| frame = await asyncio.to_thread(self._get_screen, sct, monitor)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\AppData\Local\Programs\Python\Python312\Lib\asyncio\threads.py", line 25, in to_thread
| return await loop.run_in_executor(None, func_call)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\AppData\Local\Programs\Python\Python312\Lib\concurrent\futures\thread.py", line 58, in run
| result = self.fn(*self.args, **self.kwargs)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\live_api_starter.py", line 128, in _get_screen
| i = sct.grab(monitor)
| ^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\Lib\site-packages\mss\base.py", line 101, in grab
| screenshot = self._grab_impl(monitor)
| ^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\Lib\site-packages\mss\windows.py", line 226, in _grab_impl
| srcdc, memdc = self._handles.srcdc, self._handles.memdc
| ^^^^^^^^^^^^^^^^^^^
| AttributeError: '_thread._local' object has no attribute 'srcdc'

Any idea what the cause could be?

Reply:
This could be the reason: https://stackoverflow.com/a/76600858

Reply:
With this fix https://github.com/google-gemini/cookbook/pull/359/commits/44c4a95433691d9e0b4457b7abdf8c7dcdf0335e - works?